core/core.go

```diff
@@ -498,12 +507,17 @@ func (n *IpfsNode) startOnlineServicesWithHost(ctx context.Context, host p2phost
 		n.PubSub = service
 	}
 
-	// setup routing service
-	r, err := routingOption(ctx, host, n.Repo.Datastore(), n.RecordValidator)
-	if err != nil {
-		return err
```

Comment:
1. This comment needs to explain what tests need this and why.
2. I'm quite positive there's a non-hacky way to do this.